src/components/avatar/utils.ts

```diff
@@ -1,24 +1,39 @@
 import { RGB } from "../color-picker/interfaces";
 import { hexToRGB } from "../color-picker/utils";
+import { stringArgs } from "./interfaces";
+import Color from "color";
 
 /**
  * Convert a string to a valid hex by hashing its contents
  * and using the hash as a seed for three distinct color values
  *
- * @param str
+ * @param name {string}
+ * @param theme {string}
  */
-export function stringToHex(str: string): string {
-  let hash = 0;
-  for (let i = 0; i < str.length; i++) {
-    hash = str.charCodeAt(i) + ((hash << 5) - hash);
+
+export function stringToHexCompliant(args: stringArgs): string {
+  const { id, name, theme } = args;
+  let hex;
+
+  if (id) {
+    hex = id;
+  } else {
+    //returns a 6-digit color code based on a string input
+    const hash = Array.from(name).reduce((acc, char) => char.charCodeAt(0) + ((acc << 5) - acc), 0);
```

Comment:
Can you explain these changes (compared to the original implementation) and why they are needed?